### PR TITLE
fix: replace the usage of u/snake-key with u/->snake_case_en

### LINKED ISSUE #6: syncing database stuck with metabase 0.47
sync database will stuck forever

metabase report:
```
2023-10-13 14:37:00 java.lang.NoClassDefFoundError: metabase/util$snake_key
2023-10-13 14:37:00     at metabase.driver.databend$fn__652.invokeStatic(databend.clj:427)
2023-10-13 14:37:00     at metabase.driver.databend$fn__652.invoke(databend.clj:427)
2023-10-13 14:37:00     at clojure.lang.MultiFn.invoke(MultiFn.java:234)
2023-10-13 14:37:00     at metabase.driver.databend$get_tables_in_db.invokeStatic(databend.clj:135)
2023-10-13 14:37:00     at metabase.driver.databend$fn__457.invokeStatic(databend.clj:163)
2023-10-13 14:37:00     at metabase.driver.databend$fn__457.invoke(databend.clj:160)
2023-10-13 14:37:00     at clojure.lang.MultiFn.invoke(MultiFn.java:234)
2023-10-13 14:37:00     at metabase.sync.fetch_metadata$fn__97141$db_metadata__97146$fn__97147.invoke(fetch_metadata.clj:15)
2023-10-13 14:37:00     at metabase.sync.fetch_metadata$fn__97141$db_metadata__97146.invoke(fetch_metadata.clj:12)
2023-10-13 14:37:00     at metabase.sync.sync_metadata$fn__99560$sync_db_metadata_BANG___99565$fn__99566.invoke(sync_metadata.clj:58)
2023-10-13 14:37:00     at metabase.sync.sync_metadata$fn__99560$sync_db_metadata_BANG___99565.invoke(sync_metadata.clj:55)
2023-10-13 14:37:00     at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invokeStatic(sync_databases.clj:75)
2023-10-13 14:37:00     at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invoke(sync_databases.clj:66)
2023-10-13 14:37:00     at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:85)
2023-10-13 14:37:00     at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
2023-10-13 14:37:00     at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
2023-10-13 14:37:00 2023-10-13 06:37:00,127 ERROR core.ErrorLogger :: Job (DEFAULT.metabase.task.sync-and-analyze.job threw an exception.
```

### PR DESCRIPTION
**Pull request by washanhanzi**
Fix #6 

I know nothing about everything.

Copy paste from this [PR](https://github.com/ClickHouse/metabase-clickhouse-driver/pull/160)

And then use the github action to build myself a driver, it seems work.

### CODE REVIEW

**Comment by hantmac** [comment]
Thanks for your contribution! It seems that change the node version may cause the CI failed.https://github.com/databendcloud/metabase-databend-driver/actions/runs/6507280978/job/17696630460?pr=7#step:10:15

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v2
         with:
-          node-version: "16"
+          node-version: "18"
```
Comment (hantmac) [inline .github/workflows/check.yaml]: Why need node 18?

Comment (washanhanzi) [reply .github/workflows/check.yaml]: I don't know. Just copied from the clickhouse PR😂.

Comment (hantmac) [reply .github/workflows/check.yaml]: Try to keep node 16 and wait the CI 😄

Comment (washanhanzi) [reply .github/workflows/check.yaml]: Hmm, I try to change it back, and it still failed in my fork repo. https://github.com/washanhanzi/metabase-databend-driver/actions/runs/6515055462/job/17697018583

**Comment by washanhanzi** [comment]
I changed node to 16

File: .github/workflows/check.yaml (hunk 1 above)
Comment (washanhanzi) [reply .github/workflows/check.yaml]: I don't know what is `build-static-viz` doing. With version 18 the fork repo action failed so I just comment it and build the driver.

Comment (washanhanzi) [reply .github/workflows/check.yaml]: Hmm, the upstream change it https://github.com/metabase/metabase/commit/dbd4fbaac7a6c8fd626e7604d5dee47a59e27647

Comment (hantmac) [reply .github/workflows/check.yaml]: Oh I got it. Just merge this pr first and I will adjust the CI yaml. Thanks!

Comment (washanhanzi) [reply .github/workflows/check.yaml]: Ok, thanks.

**Pull request merged**